### PR TITLE
(FACT-1284) Expose pid from execute result; Add rb_last_status_set binding

### PR DESCRIPTION
**Pull request by branan**
These are needed for FACT-1284, in order to expose the process exit status to the ruby API

### CODE REVIEW

File: ruby/inc/leatherman/ruby/api.hpp [hunk 1]
```diff
@@ -376,6 +382,11 @@ namespace leatherman {  namespace ruby {
         VALUE (* const rb_require)(char const*);
 
         /**
+         * Intern MRI method. We're being naughty
```
Comment (kylog) [inline ruby/inc/leatherman/ruby/api.hpp]: Gulp. But yes, we are needing an internal method here ....

**Comment by MikaelSmith** [comment]
Any spec tests we could modify? I'm pretty sure we have a couple we could check for zero/non-zero pids.

**Comment by branan** [comment]
@MikaelSmith since we have an explicit constructor in the result struct which requires the pid, we get compiler errors if pid isn't passed in (at least we do on GCC 5.3). That's how I caught the "no executable" cases.

**Comment by branan** [comment]
In fact I'd wager (though I'd have to double-check the spec) that since we have a constructor those default values are meaningless

**Pull request merged**